src/codeu/chat/server/Server.java

```diff
@@ -206,7 +246,125 @@ public void run() {
         timeline.scheduleIn(RELAY_REFRESH_MS, this);
       }
     });
-    
+  }
+
+  // This method will write the contents of dataList to a txt file
+  public void saveServer() {
```

Comment:
Please consider writing tests for `saveServer()` and `restoreServer()` to verify they're working as expected.